Tests/SentryProfilerTests/SentryProfilerSwiftTests.swift

```diff
@@ -715,36 +666,31 @@ private extension SentryProfilerSwiftTests {
         Dynamic(hub).tracesSampler.random = TestRandom(value: 1.0)
 
         let span = try fixture.newTransaction()
-        let exp = expectation(description: "Span finishes")
-        DispatchQueue.global().asyncAfter(deadline: .now() + 2.0) {
-            span.finish()
+        forceProfilerSample()
+        fixture.currentDateProvider.advance(by: 5)
+        span.finish()
 
-            guard let client = self.fixture.client else {
-                XCTFail("Expected a valid test client to exist")
+        guard let client = self.fixture.client else {
+            XCTFail("Expected a valid test client to exist")
+            return
+        }
+
+        switch expectedDecision {
+        case .undecided, .no:
+            guard let event = client.captureEventWithScopeInvocations.first else {
+                XCTFail("Expected to capture at least 1 event, but without a profile")
                 return
             }
-
-            switch expectedDecision {
-            case .undecided, .no:
-                guard let event = client.captureEventWithScopeInvocations.first else {
-                    XCTFail("Expected to capture at least 1 event, but without a profile")
-                    return
-                }
-                XCTAssertEqual(0, event.additionalEnvelopeItems.count)
-            case .yes:
-                guard let event = client.captureEventWithScopeInvocations.first else {
-                    XCTFail("Expected to capture at least 1 event with a profile")
-                    return
-                }
-                XCTAssertEqual(1, event.additionalEnvelopeItems.count)
-            @unknown default:
-                fatalError("Unexpected value for sample decision")
+            XCTAssertEqual(0, event.additionalEnvelopeItems.count)
+        case .yes:
+            guard let event = client.captureEventWithScopeInvocations.first else {
+                XCTFail("Expected to capture at least 1 event with a profile")
+                return
             }
-
-            exp.fulfill()
+            XCTAssertEqual(1, event.additionalEnvelopeItems.count)
+        @unknown default:
+            fatalError("Unexpected value for sample decision")
```

Comment:
`m`: I would like to avoid `fatalError` in tests, as they will cause the test suite to crash. Please use `XCTFail` instead.

Reply:
Yep, good call!